Vendor mode uses trucks owned by current user

nockertsb opened the pull request:
@MariusVolkhart This includes the changes necessary to update serving mode and load the menu for the proper truck when a user signs in. After signing in the getVendorTrucks API route is called and the appropriate trucks are marked as 'owned_by_current_user' in the database. In the future we can have some sort of context dropdown to switch between trucks if there is more that one, but right now it just grabs the first one. When a vendor logs out that column gets reset on the appropriate trucks.

MariusVolkhart commented on mobile/src/main/java/com/truckmuncher/truckmuncher/vendor/VendorHomeActivity.java:
```diff
@@ -111,4 +133,63 @@ public void run() {
+                .where(PublicContract.Truck.OWNED_BY_CURRENT_USER, EQUALS, 1)
+                .build();
+        String[] projection = TrucksOwnedByUserQuery.PROJECTION;
+        Uri uri = Contract.TRUCK_PROPERTIES_URI;
```
Since it's a query, use `PublicContract.TRUCK_URI`. The internal ones in the `Contract` class are only there b/c we can't do a write against a SQL view

MariusVolkhart replied:
Actually ignore this comment for now.

MariusVolkhart commented on mobile/src/main/java/com/truckmuncher/truckmuncher/vendor/VendorHomeActivity.java:
```diff
@@ -13,33 +18,47 @@
+        VendorMapFragment.OnMapLocationChangedListener, VendorHomeFragment.OnServingModeChangedListener,
+        LoaderManager.LoaderCallbacks<Cursor> {
 
     public static final String USERNAME = "VendorHomeActivity.username";
```
Unused. delete

MariusVolkhart commented on mobile/src/main/java/com/truckmuncher/truckmuncher/vendor/VendorTrucksService.java:
```diff
@@ -66,6 +66,7 @@ protected void onHandleIntent(Intent intent) {
                 values.put(PublicContract.Truck.KEYWORDS, Contract.convertListToString(truck.keywords));
                 values.put(PublicContract.Truck.COLOR_PRIMARY, truck.primaryColor);
                 values.put(PublicContract.Truck.COLOR_SECONDARY, truck.secondaryColor);
+                values.put(PublicContract.Truck.OWNED_BY_CURRENT_USER, 1);
```
put `true` instead of 1. Easier to read and the conversion will happen automatically.

MariusVolkhart commented on mobile/src/main/java/com/truckmuncher/truckmuncher/vendor/menuadmin/ResetVendorTrucksService.java:
```diff
@@ -0,0 +1,37 @@
+            contentValues[i] = (ContentValues) parcelables[i];
+        }
+
+        getContentResolver().bulkInsert(Contract.TRUCK_PROPERTIES_URI, contentValues);
```
Instead of a bulk insert, do a batch of updates. That way, you don't need to pull all the data out of the db and put it all in again. 

`getContentResolver().applyBatch()`. There's a pretty good write up available about it here http://www.grokkingandroid.com/better-performance-with-contentprovideroperation/

Then the `IntentService` here could even just take a list/array of ids for the db instead of full `ContentValues`

MariusVolkhart commented on mobile/src/main/res/raw/truckmuncher.sql:
```diff
@@ -9,7 +9,8 @@ CREATE TABLE `truck_properties` (
 	`color_primary`	            TEXT,
-	`color_secondary`	        TEXT
+	`color_secondary`	        TEXT,
+	`owned_by_current_user`     INTEGER DEFAULT 0
```
I feel like this would be more appropriate on the state rather than a property. Reason being, if we expose the data to other apps, it would be weird to have this change just because a user logged out in our app. Also, think it should be added to the `Truck` view?

nockertsb commented:
I moved the column to the truck_state table and changed a few other things as needed. I ended up moving two files to a different package, because I put them in the wrong one to begin with.

@MariusVolkhart please review

MariusVolkhart commented:
:+1: on the changes, but they broke a test (correctly)
